memory/src/main/scala/filodb.memory/BlockManager.scala

```diff
@@ -186,6 +189,10 @@ class PageAlignedBlockManager(val totalMemorySizeInBytes: Long,
       if (entry.getValue.isEmpty) timeOrderedListIt.remove()
     }
     if (reclaimed < num) reclaimFrom(usedBlocks, stats.blocksReclaimedMetric)
+    if (reclaimed < num) {
```

Comment:
Worth adding a comment that goes like " // if we do not get required blocks even after reclaim call"
There would be a tendency to combine both if blocks to one.